doc/ARS/Local_ARS_HLD.md

```diff
@@ -0,0 +1,892 @@
+
+#### PortsOrch
+portsorch handles all ports-related configurations. New functionality for enabling ARS on ports.
+
+#### AclOrch
+aclorch is used to deal with configurations of ACL table and ACL rules. New rule action for control ARS operation.
+
+
+ARS is closely tied to vendor-specific hardware implementations, requiring thorough validation of all configurations using SAI capabilities.
+
+### High-Level Design 
+
+- A standard ARS configuration consists of creating an ARS profile and enabling ARS on the designated ports. This process also includes activating ARS for nexthop groups (Adaptive Routing) and LAGs (Adaptive Switching). At present, configurations are carried out manually via CONFIG_DB, with potential future enhancements to incorporate management through an external controller or routing protocol extensions.
+
+
+- New orchagent: ArsOrch responsible for 
+1. Creating/Updating ARS profile 
+2. Enable interfaces for ARS
+3. Enable ARS over nexthop group
+4. Allow configuration of path metrics
+
+- ACL changes<br>
+When a new ACL table is created, SAI needs to receive a list of supported actions which the rules belonging to this table are allowed to use.
+To support the new ARS disable action, the custom table types table schema will be extended with a ARS action attribute - "ARS_ACTION" for the actions attribute field.
+
+The diagrams below illustrate the typical sequences for ARS configuration, showcasing key workflows for both Adaptive Routing and Adaptive Switching.
+
+#### Sequence diagrams
+
+__Figure 4: Initilization flow__
+![](images/init_seq.png)
+
+1. Bind to Events
+The ArsOrch component binds to relevant events to start monitoring and processing ARS-related interfaces.
+
+2. Get ARS SAI Capabilities
+ArsOrch retrieves ARS capabilities from the SAI layer and saves them in STATE_DB.
+
+3. Add ARS_PROFILE Table
+The CONFIG_DB adds the ARS_PROFILE table, which defines the ARS configuration parameters.
+ArsOrch uses this to create the ARS SAI profile.
+
+4. Add ARS_INTERFACE Table
+The CONFIG_DB adds the ARS_INTERFACE table, causing ArsOrch to enable ARS on the corresponding ports.
+
+__Figure 5: Interface update flow__
+![](images/if_update.png)
+
+1. Link UP Notification
+The PortsOrch component detects a link state change and notifies ArsOrch when a port transitions to the "UP" state.
+
+2. Set Port Scaling Factor
+Based on the port's speed, PortsOrch determines the appropriate scaling factor.
+The scaling factor is then set in the SAI layer through the syncd component to adjust the port's behavior accordingly.
+
+__Figure 6: ARS Nexthop group table creation flow__
+![](images/ars_nhg_create.png)
+
+1. Add ARS_NEXTHOP_GROUP Table
+    * The CONFIG_DB adds the ARS_NEXTHOP_GROUP table, triggering the process.
+2. Create SAI ARS Object
+    * The ArsOrch component receives the update and creates the SAI ARS object.

```

Comment:
does design propose the creation of ARS Object per entry in ARS_NEXTHOP_GROUP table ? How to re-use the ARS Object ? SAI APIs doesn't limit the re-use